### DISCUSSION TITLE
failed to fetch internal transactions for blocks

### DISCUSSION THREAD

**Discussion by 0xA2618** [blockscout/blockscout#9573] [Q&A] [answered]
{"time":"2024-03-07T13:20:43.605Z","severity":"error","message":"failed to fetch internal transactions for blocks: ** (ErlangError) Erlang error: :nxdomain","metadata":{"count":10,"error_count":10,"fetcher":"internal_transaction"}} 

I cannot find nxdomain in the relevant configuration

**Comment by GimluCom** [comment]
Please share your backend ENVs

**Reply by 0xA2618** [reply]
backend-env: [common-blockscout.env.txt](https://github.com/blockscout/blockscout/files/14531768/common-blockscout.env.txt)
[docker-compose: 
[docker-compose.yaml.txt](https://github.com/blockscout/blockscout/files/14531822/docker-compose.yaml.txt)


I am started in docker-compose mode, where there is a subnet configuration in explorer-backend-l2, which I have tried to adjust but failed. Attached are deployment configurations of backend.env and docker-compose

thanks

**Reply by GimluCom** [reply]
Looks like your hostname can't be resolved inside the docker container. Try using a IP/domain or `host.docker.internal` if the RPC is on your host machine.

**Comment by 0xA2618** [comment]
1. BlockScout v6.0.0. I started a new environment on another server and checked the explorer backend l2 db (service in docker-compose.yml). In the database, when looking at the internal_transactions table, it was found to be an empty table, and this error was reported. I don't know where the data for this error came from

2. I am https://github.com/blockscout/blockscout/blob/master/cspell.json Seeing the configuration of an nxdomain, I am not sure if it is related to domain name resolution

**Comment by 0xA2618** [comment, marked as answer]
I have found the cause of my problem. The ETHEREUM_JSONRPC.HTTP_URL configured in the configuration file is not correct. 
My zkevm-node was started using binary, 
and blockScout was started using Docker Compose. 
ETHEREUM_JSONRPC.HTTP_URL can be found by using the host's internal network address
